Lib/http/cookiejar.py

```diff
@@ -532,15 +532,42 @@ def parse_ns_headers(ns_headers):
     return result
 
 
+# only kept for backwards compatibilty.
 IPV4_RE = re.compile(r"\.\d+$", re.ASCII)
+
+def _is_ipv4_hostname(text):
+    from ipaddress import IPv4Address
+    try:
+        IPv4Address(text)
+    except ValueError:
+        return False
+    return True
+
+def _is_ipv6_hostname(text):
+    if text.startswith('[') and text.endswith(']'):
+        from ipaddress import IPv6Address
+        try:
+            IPv6Address(text[1:-1])
+        except ValueError:
+            return False
+        return True
+    return False
+
+def is_ip_like_hostname(text):
+    """Return True if text is a valid hostname in the form of IP address.
+
+    A valid IP-like hostname is either an IPv4 address or
+    an IPv6 enclosed in brackets (for instance, "[::1]").
+    """
+    if _is_ipv4_hostname(text) or _is_ipv6_hostname(text):
+        return True
+    return False
 def is_HDN(text):
```

Comment:
```suggestion
    return False

def is_HDN(text):
```

Let's add some blank lines. We've added enough functions I think.